scripts/extract_translation_templates.py

```diff
@@ -10,125 +10,136 @@
 
-        if not contents_match:
-            diff_results = difflib.unified_diff(
-                source_contents,
-                target_contents,
-                fromfile=source_file,
-                tofile=target_file,
-            )
-            logger.info("".join(list(diff_results)))
+    return contents_match
 
-        print_filename_results(f"{source_file}", contents_match)
 
-        return contents_match
+def build_schema_templates(output_dir):
 
-    def build_schema_templates(output_dir):
+    for schema_name in SCHEMAS_TO_EXTRACT:
+        template_file = f"{schema_name}.pot"
+        schema_file = f"{schema_name}.json"
 
-        for schema_name in SCHEMAS_TO_EXTRACT:
-            template_file = f"{schema_name}.pot"
-            schema_file = f"{schema_name}.json"
+        logger.info("Building %s/%s", output_dir, template_file)
 
-            logger.info("Building %s/%s", output_dir, template_file)
+        handle_extract_template(f"schemas/en/{schema_file}", output_dir)
 
-            handle_extract_template(f"schemas/en/{schema_file}", output_dir)
+        logger.info("Built %s/%s", output_dir, template_file)
 
-            logger.info("Built %s/%s", output_dir, template_file)
 
-    def check_schema_templates(source_dir, target_dir):
-        return all(
-            compare_files(source_dir, target_dir, f"{schema_name}.pot")
-            for schema_name in SCHEMAS_TO_EXTRACT
-        )
+def check_schema_templates(source_dir, target_dir):
+    return all(
+        compare_files(source_dir, target_dir, f"{schema_name}.pot")
+        for schema_name in SCHEMAS_TO_EXTRACT
+    )
 
-    if __name__ == "__main__":
-        parser = argparse.ArgumentParser(
-            description="Extract translation templates from runner"
-        )
-        parser.add_argument(
-            "--test",
-            help="Test the templates without making changes",
-            action="store_true",
-        )
 
-        args = parser.parse_args()
+if __name__ == "__main__":
 
-        if args.test:
-            with tempfile.TemporaryDirectory(dir="/tmp") as temp_dir:
-                build_schema_templates(temp_dir)
+    page = requests.get("https://github.com/ONSdigital/eq-translations/commit/master/")
 
-                success = check_schema_templates("translations", temp_dir)
 
-                if not success:
-                    logger.error(
-                        "Translation templates are not up to date. Run make translation-templates to fix this"
-                    )
-                    sys.exit(1)
+    def get_ref():
+        f = open("Pipfile.lock", "r")
+        file = f.readlines()
+        for i, line in enumerate(file):
+            if '"eq-translations"' in line:
+                ref = re.compile(r"\w{40}")
+                mo = ref.search(file[i + 3])
+                return mo.group()
+        print("eq-translations not in Pipfile.lock")
+
+
+    if get_ref() in page.text:
```

Comment:
This fails when the commit is the second to last commit. It would still be on the page

Reply:
Is this solved now when we use GitHub api to check "sha"?